api/graphql/mutations/network.js

```diff
@@ -2,12 +2,49 @@ import validateNetworkData from '../../models/network/validateNetworkData'
 import underlyingUpdateNetwork from '../../models/network/updateNetwork'
 import convertGraphqlData from './convertGraphqlData'
 
-export function updateNetwork (userId, { id, data }) {
+// TODO: more integrated `isAdmin` handling for mutations?
+export function networkMutationPermissionCheck ({ userId, isAdmin = false }, networkId) {
```

Comment:
feel free to use async/await btw; i think that would make this simpler?

Reply:
I saw you started using that with `makeModels`! I'll give it a crack. I also realised I need to check `hasModeratorRole` for the new mod to avoid dups.